### PR TITLE
DOC: Add rolling in reference/groupby.rst

### LINKED ISSUE #51097: DOC: GroupBy.rolling() not in docs
### Pandas version checks

- [X] I have checked that the issue still exists on the latest versions of the docs on `main` [here](https://pandas.pydata.org/docs/dev/)

### Location of the documentation

https://pandas.pydata.org/docs/reference/groupby.html
https://pandas.pydata.org/docs/dev/reference/groupby.html

### Documentation problem

In the source code for `GroupBy` I see [rolling](https://github.com/pandas-dev/pandas/blob/main/pandas/core/groupby/groupby.py#L2781) and `ewm` and `expanding` but I don't see these in the docs.

I can't work out if they've been deprecated, or are very new, or if there's a bug in the documentation creation.

### Suggested fix for documentation

Depending on why they're not in the docs, either an update to the docstring to mention that these aren't part of the official API (and why, and what to use instead maybe, or maybe a deprecation warning) or update the docs to include them.

### PR DESCRIPTION
**Pull request by luke396**
- [ ] close #51097

### CODE REVIEW

**Comment by mroeschke** [comment]
Since we have docstring checks, the docstrings of these will need to be improved in this PR

```
Error: /home/runner/work/pandas/pandas/pandas/core/groupby/groupby.py:2787:RT01:pandas.core.groupby.DataFrameGroupBy.rolling:No Returns section found
Error: /home/runner/work/pandas/pandas/pandas/core/groupby/groupby.py:2787:RT01:pandas.core.groupby.SeriesGroupBy.rolling:No Returns section found
Error: /home/runner/work/pandas/pandas/pandas/core/groupby/groupby.py:2787:EX01:pandas.core.groupby.DataFrameGroupBy.rolling:No examples section found
Error: /home/runner/work/pandas/pandas/pandas/core/groupby/groupby.py:2787:EX01:pandas.core.groupby.SeriesGroupBy.rolling:No examples section found
```

**Comment by luke396** [comment]
> Since we have docstring checks, the docstrings of these will need to be improved in this PR
> 
> ```
> Error: /home/runner/work/pandas/pandas/pandas/core/groupby/groupby.py:2787:RT01:pandas.core.groupby.DataFrameGroupBy.rolling:No Returns section found
> Error: /home/runner/work/pandas/pandas/pandas/core/groupby/groupby.py:2787:RT01:pandas.core.groupby.SeriesGroupBy.rolling:No Returns section found
> Error: /home/runner/work/pandas/pandas/pandas/core/groupby/groupby.py:2787:EX01:pandas.core.groupby.DataFrameGroupBy.rolling:No examples section found
> Error: /home/runner/work/pandas/pandas/pandas/core/groupby/groupby.py:2787:EX01:pandas.core.groupby.SeriesGroupBy.rolling:No examples section found
> ```

I added some docstring, but they definitely need further discussion. 

The parameters section is copied from [pandas.DataFrame.rolling](https://pandas.pydata.org/docs/dev/reference/api/pandas.DataFrame.rolling.html#pandas.DataFrame.rolling), the parameter `step` is not implemented, so I removed it.  Other parameters are implemented, but the exact description I'm not sure if I need to keep changing.

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -2723,10 +2723,134 @@ def resample(self, rule, *args, **kwargs):
...
+        ...                    'B': [1, 2, 3, 4],
+        ...                    'C': [0.362, 0.227, 1.267, -0.562]})
+        >>> df
+              A  B      C
+        0     1  1  0.362
+        1     1  2  0.227
+        2     2  3  1.267
+        3     2  4 -0.562
+
+        >>> df.groupby('A').rolling(2).sum()
```
Comment (mroeschke) [inline pandas/core/groupby/groupby.py:2831]: These examples should show calling groupby rolling specifically

Comment (luke396) [reply pandas/core/groupby/groupby.py:2831]: Sorry, I didn't fully understand what changes should be made to it. Do you mean that these examples do not show the usefulness of groupy rolling and that the data should be reorganized?

Comment (mroeschke) [reply pandas/core/groupby/groupby.py:2831]: Ah sorry my mistake. The examples look good

File: pandas/core/groupby/groupby.py [hunk 2]
```diff
@@ -2723,10 +2723,134 @@ def resample(self, rule, *args, **kwargs):
 
     @final
     @Substitution(name="groupby")
```
Comment (mroeschke) [inline pandas/core/groupby/groupby.py:2725]: I think this can be removed too

Comment (luke396) [reply pandas/core/groupby/groupby.py]: Fine, how about this time?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @luke396

**Pull request merged**